### PR TITLE
fix: theming toast component

### LINKED ISSUE #17878: Theming the toast
# Problem

The current toast implementation needs an `:icon-color` which needs to be provided along with the current app theme.
The toast should read the theme as other components in the app: using our theme HOC.

Example in  `status-im2.contexts.wallet.edit-account.view`:

```clojure
(rf/dispatch [:toasts/upsert
              {:id         :edit-account
               :icon       :i/correct
               :icon-color (colors/resolve-color :success theme)
               :text       (i18n/label message)}])
```

## Acceptance Criteria
App is working as expected and the toast is showing in the expected ways for every screen,

## Notes

Related issue:
- https://github.com/status-im/status-mobile/issues/16878

### PR DESCRIPTION
**Pull request by codemaster115**
fixes #17878 

### Summary

Refactor quo/toast component
- remove `:icon-color` and add `:type` property
- fix everywhere using toast component

#### Platforms

- Android
- iOS
- macOS
- Linux
- Windows

##### Functional

- 1-1 chats
- public chats
- group chats
- wallet / transactions
- dapps / app browsing
- account recovery
- new account
- user profile updates
- networks
- mailservers
- fleet
- bootnodes

status: ready

### CODE REVIEW

File: src/status_im/communities/core.cljs [hunk 1]
```diff
@@ -99,10 +98,10 @@
           (assoc-in db [:communities/my-pending-requests-to-join community-id] request)
           (and (= constants/community-request-to-join-state-accepted state) (not deleted))
           (do (re-frame/dispatch [:toasts/upsert
-                                  {:icon       :i/correct
-                                   :id         :joined-community
-                                   :icon-color (:positive-01 @colors/theme)
-                                   :text       (i18n/label :t/joined-community {:community name})}])
+                                  {:icon :i/correct
```
Comment (J-Son89) [inline src/status_im/communities/core.cljs]: @codemaster115 do you know does the icon ever change? or is it always `:i/correct` for type `:positive`, `:i/info` for `:negative` etc? would be nice to encapsulate that in the component if so 👍

File: src/quo/components/notifications/toast/style.cljs [hunk 2]
```diff
@@ -38,8 +38,14 @@
...
-  [theme]
-  {:color           (colors/theme-colors colors/white colors/neutral-100 theme)
+  [type theme]
+  {:color           (condp = type
+                      :negative (colors/resolve-color :danger theme)
+                      :positive (colors/resolve-color :success theme)
+                      (colors/theme-colors colors/white-opa-40
+                                           colors/neutral-80-opa-40
+                                           theme)
+                    )
```
Comment (ulisesmac) [inline src/quo/components/notifications/toast/style.cljs]: Please fix the trailing parenthesis. Also this `condp` could perfectly  be a `case`, which is simper

File: src/quo/components/notifications/toast/style.cljs [hunk 3]
```diff
@@ -38,8 +38,14 @@
   {:color (colors/theme-colors colors/white colors/neutral-100 theme)})
 
 (defn icon
-  [theme]
-  {:color           (colors/theme-colors colors/white colors/neutral-100 theme)
+  [type theme]
```
Comment (ulisesmac) [inline src/quo/components/notifications/toast/style.cljs]: Even though we are receiving a type, I'd suggested naming that var  differently (e.g. `icon-type` or whatever suits better). The reason is `type` is shadowing `clojure.core/type`.

File: src/quo/components/notifications/toast/view.cljs [hunk 4]
```diff
@@ -69,15 +69,17 @@
 (def ^:private toast-container (quo.theme/with-theme toast-container-internal))
 
 (defn toast
-  [{:keys [icon icon-color title text action undo-duration undo-on-press container-style theme user]}]
+  "Options:
+
```
Comment (ulisesmac) [inline src/quo/components/notifications/toast/view.cljs:73]: Extra empty line

**Review by ulisesmac — COMMENTED** [review]
Thanks for this PR @codemaster115 !

I left some comments

File: src/quo/components/notifications/toast/view.cljs [hunk 5]
```diff
@@ -69,15 +69,21 @@
...
-  [{:keys [icon icon-color title text action undo-duration undo-on-press container-style theme user]}]
-  (let [context-theme (or theme (quo.theme/get-theme))]
+  "Options:
+   
+   :type => :neutral/:negative/:positive
+   "
+  [{:keys [type title text action undo-duration undo-on-press container-style theme user]
+    :or   {type :neutral}}]
+  (let [context-theme (or theme (quo.theme/get-theme))
+        icon-name     (case type
```
Comment (flexsurfer) [inline src/quo/components/notifications/toast/view.cljs:79]: it should be icon-key, and keys must be with `:i/` namespace

File: src/quo/components/notifications/toast/view.cljs [hunk 6]
```diff
@@ -69,15 +69,21 @@
...
+                        :positive "incorrect"
```
Comment (flexsurfer) [inline src/quo/components/notifications/toast/view.cljs]: :i/incorrect

**Review by flexsurfer — APPROVED** [review]

File: src/quo/components/notifications/toast/view.cljs [hunk 7]
```diff
@@ -69,15 +69,21 @@
...
+   
+   :type => :neutral/:negative/:positive
+   "
+  [{:keys [type title text action undo-duration undo-on-press container-style theme user]
+    :or   {type :neutral}}]
+  (let [context-theme (or theme (quo.theme/get-theme))
+        icon-name     (case type
+                        :positive :i/incorrect
+                        :negative :i/correct
+                        :neutral  :i/placeholder)]
```
Comment (J-Son89) [inline src/quo/components/notifications/toast/view.cljs]: oh it seems unlikely that we are using :i/placeholder as the neutral icon and this is something that can be overwritten? - can you verify with design how this should be done? 🙏

**Review by J-Son89 — APPROVED** [review]
nice work @codemaster115  :)

File: src/quo/components/notifications/toast/view.cljs (hunk 7 above)
Comment (codemaster115) [reply src/quo/components/notifications/toast/view.cljs]: I confirmed that it's the placeholder.

**Review by ulisesmac — APPROVED** [review]

File: src/quo/components/notifications/toast/view.cljs (hunk 7 above)
Comment (J-Son89) [reply src/quo/components/notifications/toast/view.cljs]: I just saw that message. They said it depends on context, so for neutral it can change 👍

**Review by smohamedjavid — APPROVED** [review]
Nice work! @codemaster115 🚀

**Comment by churik** [comment]
### ISSUE 1: icons are not shown or wrong for some types of toasts

1) Toast of contact request:
image
Design: https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=1686-34385&mode=design&t=RYEJ3yD8xFuwVafU-0

2) Toast of mark all as read in Activity Centre:
IMAGE 2023-11-30 16:47:29
Figma: https://www.figma.com/file/eDfxTa9IoaCMUy5cLTp0ys/Shell-for-Mobile?type=design&node-id=5387-529829&mode=design&t=CZJtcmWLiKfqVhVd-4

3) Mute toast
IMAGE 2023-11-30 16:59:30

In [design system](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=3928-77614&mode=design&t=Hj4OgPEQjgP7gMiU-4) there Success is with enabled green checkmark.


**OS:** IOS, Android

**Comment by J-Son89** [comment]
@codemaster115 is there any update on this pr? 🤔 I'm working on some colors related issue and would benefit from having this issue addressed 👍

**Comment by codemaster115** [comment]
> @codemaster115 is there any update on this pr? 🤔 I'm working on some colors related issue and would benefit from having this issue addressed 👍

Sorry, I still didn't have a chance to resolve. Will update soon.

**Comment by codemaster115** [comment]
@J-Son89 I've updated this PR so please review.

File: src/quo/components/notifications/toast/view.cljs (hunk 7 above)
Comment (J-Son89) [reply src/quo/components/notifications/toast/view.cljs]: @codemaster115 please review this comment

Comment (codemaster115) [reply src/quo/components/notifications/toast/view.cljs]: @J-Son89 It's resolved now.

**Comment by codemaster115** [comment]
> ### ISSUE 1: icons are not shown or wrong for some types of toasts
> 1. Toast of contact request:
>    image
>    Design: https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=1686-34385&mode=design&t=RYEJ3yD8xFuwVafU-0
> 2. Toast of mark all as read in Activity Centre:
>    IMAGE 2023-11-30 16:47:29
>    Figma: https://www.figma.com/file/eDfxTa9IoaCMUy5cLTp0ys/Shell-for-Mobile?type=design&node-id=5387-529829&mode=design&t=CZJtcmWLiKfqVhVd-4
> 3. Mute toast
>    IMAGE 2023-11-30 16:59:30
> 
> In [design system](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=3928-77614&mode=design&t=Hj4OgPEQjgP7gMiU-4) there Success is with enabled green checkmark.
> 
> **OS:** IOS, Android

@churik All issues are resolved now.

**Comment by qoqobolo** [comment]
Hey @codemaster115, could you resolve the conflicts please?
I'll retest PR after that, thanks!

**Comment by codemaster115** [comment]
> Hey @codemaster115, could you resolve the conflicts please? I'll retest PR after that, thanks!

@qoqobolo All conflicts are resolved with up-to-date.

**Comment by qoqobolo** [comment]
Thanks for your work @codemaster115 , LGTM!
Please resolve another conflict and let me know if after that PR will need to be tested one more time. 
If this is not required, it can be merged.

**Comment by qoqobolo** [comment]
Hey @codemaster115 , any updates on it?🙂
Does it need to be retested or ready for merge?

**Comment by codemaster115** [comment]
> Hey @codemaster115 , any updates on it?🙂 Does it need to be retested or ready for merge?

No need to retest it. I will rebase and merge it soon.

**Pull request merged**